### PR TITLE
✨ feat: add system agent to select another model provider for translation

### PR DESCRIPTION
**Pull request by phsontung**
#### 💻 变更类型 | Change Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] ⚡️ perf
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

Screenshot from 2024-04-05 08-22-06

Screenshot from 2024-04-05 08-26-49

### CODE REVIEW

**Comment by arvinxx** [comment]
@phsontung hello, sorry for the later reply. Please rebase the config.

**Comment by phsontung** [comment]
> @phsontung hello, sorry for the later reply. Please rebase the config.

I have synced with main branch and update source code. Please help review bro.

**Comment by arvinxx** [comment]
@phsontung I think it's really a great idea to have system agent config for LobeChat. but I prefer a single tab Named "Agent Config" for this. So we can set different models to summarize, translate, auto agent meta  and so on. And in this PR just adding translate config is fine.

File: src/const/settings/index.ts [hunk 1]
```diff
@@ -151,6 +153,13 @@ export const DEFAULT_TTS_CONFIG: GlobalTTSConfig = {
   sttServer: 'openai',
 };
 
+export const DEFAULT_TRANSLATION_CONFIG: GlobalTranslationConfig = {
+  model: DEFAULT_AGENT_CONFIG.model,
+  provider: DEFAULT_AGENT_CONFIG.provider || ModelProvider.OpenAI,
```
Comment (arvinxx) [inline src/const/settings/index.ts]: Just set as OpenAI

File: src/store/chat/slices/enchance/action.ts [hunk 2]
```diff
@@ -48,12 +53,19 @@ export const chatEnhance: StateCreator<
     topicId: get().activeTopicId,
     ...data,
   }),
+  getCurrentTranslationSetting: () => {
```
Comment (arvinxx) [inline src/store/chat/slices/enchance/action.ts]: It seems this method is not used in somewhere else? maybe we can remove it?

**Pull request merged**